src/stash-debug/src/main.rs

```diff
@@ -0,0 +1,282 @@
+            Self::Catalog => vec![
+                catalog::COLLECTION_CONFIG.name().to_string(),
+                catalog::COLLECTION_SETTING.name().to_string(),
+                catalog::COLLECTION_ID_ALLOC.name().to_string(),
+                catalog::COLLECTION_SYSTEM_GID_MAPPING.name().to_string(),
+                catalog::COLLECTION_COMPUTE_INSTANCES.name().to_string(),
+                catalog::COLLECTION_COMPUTE_INTROSPECTION_SOURCE_INDEX
+                    .name()
+                    .to_string(),
+                catalog::COLLECTION_COMPUTE_REPLICAS.name().to_string(),
+                catalog::COLLECTION_DATABASE.name().to_string(),
+                catalog::COLLECTION_SCHEMA.name().to_string(),
+                catalog::COLLECTION_ITEM.name().to_string(),
+                catalog::COLLECTION_ROLE.name().to_string(),
+                catalog::COLLECTION_TIMESTAMP.name().to_string(),
+                catalog::COLLECTION_SYSTEM_CONFIGURATION.name().to_string(),
+                catalog::COLLECTION_AUDIT_LOG.name().to_string(),
+                catalog::COLLECTION_STORAGE_USAGE.name().to_string(),
+            ],
+            Self::Storage => vec![
+                storage::METADATA_COLLECTION.name().to_string(),
+                storage::METADATA_EXPORT.name().to_string(),
+            ],
+        })
+    }
+
+    async fn dump(
+        &self,
+        stash: &mut impl Stash,
+    ) -> Result<BTreeMap<&str, serde_json::Value>, anyhow::Error> {
+        macro_rules! dump_col {
+            ($col:expr) => {
+                ($col.name(), serde_json::to_value($col.iter(stash).await?)?)
+            };
+        }
+
+        let data = BTreeMap::from_iter(match self {
+            Usage::Catalog => vec![
+                dump_col!(catalog::COLLECTION_CONFIG),
+                dump_col!(catalog::COLLECTION_ID_ALLOC),
+                dump_col!(catalog::COLLECTION_SYSTEM_GID_MAPPING),
+                dump_col!(catalog::COLLECTION_COMPUTE_INSTANCES),
+                dump_col!(catalog::COLLECTION_COMPUTE_INTROSPECTION_SOURCE_INDEX),
+                dump_col!(catalog::COLLECTION_COMPUTE_REPLICAS),
+                dump_col!(catalog::COLLECTION_DATABASE),
+                dump_col!(catalog::COLLECTION_SCHEMA),
+                dump_col!(catalog::COLLECTION_ITEM),
+                dump_col!(catalog::COLLECTION_ROLE),
+                dump_col!(catalog::COLLECTION_TIMESTAMP),
+                dump_col!(catalog::COLLECTION_SYSTEM_CONFIGURATION),
+                dump_col!(catalog::COLLECTION_AUDIT_LOG),
+                dump_col!(catalog::COLLECTION_STORAGE_USAGE),
+            ],
+            Usage::Storage => {
+                let mut data = vec![dump_col!(storage::METADATA_COLLECTION)];
+                // This collection sometimes hasn't been created yet.
+                if stash
+                    .collections()
+                    .await?
+                    .contains(storage::METADATA_EXPORT.name())
```

Comment:
This is subjective, but I think that this check should actually be moved into the `dump_col!` macro somehow. Instead of relying on implementation details for each collection we can just check them all and see if they exist. This will likely make maintaining this easier in the future.

Reply:
Agreed, done.